e2e/playwright/favorite-actions/src/tests/mark-favorite.e2e.ts

```diff
@@ -0,0 +1,189 @@
+        fileNotFav3Id,
+        fileNotFav4Id
+      ]);
+    } catch (error) {
+      console.error(`beforeAll failed: ${error}`);
+    }
+  });
+
+  test.beforeEach(async ({ loginPage }) => {
+    await Utils.tryLoginUser(loginPage, username, username, 'beforeEach failed');
+  });
+
+  test.afterAll(async () => {
+    await Utils.deleteNodesSitesEmptyTrashcan(nodesApi, trashcanApi, 'afterAll failed');
+  });
+
+  test.describe('on Personal Files', () => {
+    test.afterAll(async () => {
+      try {
+        await favoritesApi.addFavoritesByIds('file', [fileFavUIId, fileFav1Id, fileFav2Id, fileFav3Id, fileFav4Id]);
+        await favoritesApi.addFavoriteById('folder', folderId);
+        await favoritesApi.removeFavoritesByIds(username, [fileNotFav1Id, fileNotFav2Id, fileNotFav3Id, fileNotFav4Id]);
+      } catch (error) {}
+    });
+
+    test.beforeEach(async ({ personalFiles }) => {
+      await personalFiles.dataTable.performClickFolderOrFileToOpen(parent);
+    });
+
+    test('Favorite action has empty star icon for an item not marked as favorite', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(fileNotFavUI);
+      await personalFiles.acaHeader.clickMoreActions();
+      expect(await personalFiles.matMenu.isMenuItemVisible('Favorite')).toBe(true);
+    });
+
+    test('Favorite action has empty star icon for multiple selection of items when some are not favorite', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectMultiItem(fileNotFavUI, fileFavUI);
+      await personalFiles.acaHeader.clickMoreActions();
+      expect(await personalFiles.matMenu.isMenuItemVisible('Favorite')).toBe(true);
+    });
+
+    test('Favorite action has full star icon for items marked as favorite', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(fileFavUI);
+      await personalFiles.acaHeader.clickMoreActions();
+      expect(await personalFiles.matMenu.isMenuItemVisible('Remove Favorite')).toBe(true);
+    });
+
+    test('favorite a file', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(fileNotFav1);
+      await personalFiles.acaHeader.clickMoreActions();
+      await personalFiles.matMenu.clickMenuItem('Favorite');
+      expect(await favoritesApi.isFavoriteWithRetry(username, fileNotFav1Id, { expect: true })).toBe(true);
+    });
+
+    test('favorite a folder', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(folder);
+      await personalFiles.acaHeader.clickMoreActions();
+      await personalFiles.matMenu.clickMenuItem('Favorite');
+      expect(await favoritesApi.isFavoriteWithRetry(username, folderId, { expect: true })).toBe(true);
+    });
+
+    test('unfavorite an item', async ({ personalFiles }) => {
+      await personalFiles.dataTable.selectItem(fileFav1);
+      await personalFiles.acaHeader.clickMoreActions();
+      await personalFiles.matMenu.clickMenuItem('Remove Favorite');
```

Comment:
Those 3 lines look similar in several test cases, maybe they can be moved to a function?